src/bin/pg_upgrade/greenplum/check_gp.c

```diff
@@ -985,3 +987,89 @@ check_views_with_removed_functions()
 	else
 		check_ok();
 }
+
+static void
+check_views_with_removed_types()
+{
+	if (GET_MAJOR_VERSION(old_cluster.major_version) > 904)
+		return;
+
+	char  output_path[MAXPGPATH];
+	FILE *script = NULL;
+	bool  found = false;
+	int   dbnum;
+	int   i_viewname;
+
+	prep_status("Checking for views with removed types");
+
+	snprintf(output_path, sizeof(output_path), "views_with_removed_types.txt");
```

Comment:
Does this need the logbase dir?

```suggestion
	snprintf(output_path, sizeof(output_path), "%s/%s,
			 log_opts.basedir, "views_with_removed_types.txt");
```

Reply:
It does. But I'm going to leave it for now. Since I've already pushed a couple of commits before we detected this issue. We'll update it at the same time the gpupgrade tests are updated.